man/rusmux.1
.TH rusmux "1" "February 10, 2026" "rusmux 1.0.0" "User Commands"
.SH NAME
rusmux \- manage your tmux project sessions

.SH SYNOPSIS
.B rusmux
.I command
[arguments]

.SH DESCRIPTION
.B rusmux
is a tool designed to manage and automate project session-based tasks with a variety of commands
for different actions involving project lifecycle management.

Features include:
.IP
- Utilize existing `.yaml` project files from tmuxinator.
.IP
- A standalone binary for easy distribution.
.IP
- Use tmux to manage various project sessions efficiently.

.SH COMMANDS
.TP
.B run
Run the project's commands (aliases: start, r).
.I Arguments:
Project name or filesystem path.
Example:
.nf
rusmux run my_project
rusmux run /path/to/project.yaml
.fi

.TP
.B stop
Stop a project's session (aliases: kill, s).
.I Arguments:
Project name or filesystem path.
Example:
.nf
rusmux stop my_project
.fi

.TP
.B debug
Output shell commands for a project.
.I Arguments:
Project name or filesystem path.
Example:
.nf
rusmux debug my_project
.fi

.TP
.B edit
Edit an existing project configuration.
.I Arguments:
Project name or filesystem path.
Ensure `$EDITOR` is set in your environment.
Example:
.nf
rusmux edit my_project
.fi

.TP
.B delete
Delete an existing project (confirmation required).
.I Arguments:
Project name or filesystem path.
Example:
.nf
rusmux delete my_project
.fi

.TP
.B new
Create a new project.
.I Arguments:
Project name or filesystem path.
Optional Flags:
\fB--blank\fR Don't use a template for the file.
Examples:
.nf
rusmux new my_project
rusmux new my_project --blank
.fi

.TP
.B list
List all existing projects in the configuration directory (alias: ls).
Example:
.nf
rusmux list
.fi

.TP
.B copy
Copy an existing project to a new one and edit it.
.I Arguments:
Existing project name or filesystem path.
New project name or filesystem path.
Example:
.nf
rusmux copy old_project new_project
.fi

.TP
.B doctor
Check your environment for potential issues, such as missing dependencies or
unset variables like `$SHELL` and `$EDITOR`.
Example:
.nf
rusmux doctor
.fi

.SH EXAMPLES
.TP
.B Using a pre-configured project
To run a project from its name of an existing `.yaml`, use:
.nf
rusmux run demo
.fi
.TP
.B Adding shell completions
To enable shell completions for `bash`:
.nf
sudo cp completions/rusmux.bash /usr/share/bash-completion/completions/rusmux
.fi
For zsh:
.nf
sudo cp completions/rusmux.zsh /usr/share/zsh/site-functions/_rusmux
.fi
For fish:
.nf
sudo cp completions/rusmux.fish /usr/share/fish/vendor_completions.d/rusmux.fish
.fi

.SH SEE ALSO
.tmux", tmuxinator....Similar



CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

## [0.8.5] - 2026-02-10

### Documentation

- Added man page

### Features

- Added `r` and `s` shortcuts for `run` and `stop`

### Miscellaneous Tasks

- Updated `dist` version

## [0.8.4] - 2025-11-04

### Bug Fixes

- Updated cargo-dist

## [0.8.3] - 2025-11-04

### Features

- Added cp alias to copy command

### Miscellaneous Tasks

- Updated runner image to Ubuntu 22.04

## [0.8.2] - 2025-11-04

### Bug Fixes

- Removed unmaintained yaml-rust

### Documentation

- Applied formatting and liniting hints to README.md
- Added to README example of running with a path to file

### Features

- Changed alias to visible_alias
- Added ls alias to list command

### Miscellaneous Tasks

- Cleaned clippy warnings

## [0.8.1] - 2025-02-25

### Documentation

- Added shell completion info to README

### Features

- Added shell completion scripts, closes #15

### Miscellaneous Tasks

- Release rusmux version 0.8.1

## [0.8.0] - 2025-02-11

### Features

- Added window root option, closes #16

### Miscellaneous Tasks

- Upgraded dependencies (colored, directories)
- Release rusmux version 0.8.0

## [0.7.0] - 2024-12-03

### Features

- [**breaking**] Changed optional run command to mandatory, fixes #14
- Added tabs and color to `doctor` command

### Miscellaneous Tasks

- Release rusmux version 0.7.0

### Refactor

- Changed Cli from builder to derive

## [0.6.1] - 2024-11-12

### Bug Fixes

- Fixed issue of cargo doc generating warning (proper sh blocks)

### Documentation

- Rephrased README, fixed typo, closes #12
- Described shell script installation method
- Added deprecation note to README

### Features

- Added warning for running without start command, see #14

### Miscellaneous Tasks

- Updated cargo dist
- Updated crates
- Release rusmux version 0.6.1

## [0.6.0] - 2024-08-27

### Bug Fixes

- Apply layout after each split, closes #10

### Features

- Added start and kill aliases (for run and stop)
- Added per-window options (set -w ....)

### Miscellaneous Tasks

- Updated dependencies
- Cleaned some clippy warnings
- Updated cargo-dist
- Release rusmux version 0.6.0

## [0.5.0] - 2024-04-02

### Features

- Added support for project files from filesystem path

### Miscellaneous Tasks

- Some clippy and formatting cleanups
- Updated cargo-dist
- Release rusmux version 0.5.0

## [0.4.0] - 2024-02-20

### Miscellaneous Tasks

- Removed homebrew
- Release rusmux version 0.4.0

## [0.3.12] - 2024-02-20

### Miscellaneous Tasks

- Removed toolchain pinning
- Release rusmux version 0.3.12

## [0.3.11] - 2024-02-20

### Miscellaneous Tasks

- Updated rust-toolchain-version
- Release rusmux version 0.3.11

## [0.3.10] - 2024-02-20

### Bug Fixes

- Allow name instead of project_name Resolves #11

### Documentation

- Added an example for pane multi-commands

### Features

- Multicommand panes

### Miscellaneous Tasks

- Update cargo-dist
- Added AUTHORS
- Updated Cargo.lock
- Updated CHANGELOG
- Release rusmux version 0.3.10

### Refactor

- Changed new project bool to an enum

## [0.3.9] - 2023-08-15

### Bug Fixes

- Set default pane width/height to 60%

### Miscellaneous Tasks

- Release rusmux version 0.3.9

## [0.3.8] - 2023-08-15

### Bug Fixes

- Added hooks to set layout Resolves #6

### Documentation

- Documented the `project_config` module
- Documented top level `error` module.
- Documented the `tmux` module
- Documented the `app` module
- Added rationale to README

### Miscellaneous Tasks

- Updated clap
- Bumped cargo-dist to 0.0.7
- Release rusmux version 0.3.8
- Release rusmux version 0.3.8

## [0.3.7] - 2023-04-23

### Bug Fixes

- Added human readable errors, resolves #5

### Documentation

- Added AUR install instructions

### Miscellaneous Tasks

- Removed MS Win target from cargo-dist
- Updated cargo-dist, added Apple silicon target
- Updated dependencies
- Added git-cliff as pre-release hook
- Release rusmux version 0.3.7

## [0.3.6] - 2023-03-28

### Bug Fixes

- Added config dir creation, resolves #4

### Features

- Implemented `stop` command and `on_project_stop` resolves #1

### Miscellaneous Tasks

- Bumped to 0.3.6

## [0.3.5] - 2023-03-21

### Documentation

- Added install instructions to README

### Features

- Implemented `copy` command, resolves #3

### Miscellaneous Tasks

- Removed windows from release build
- Bumped to 0.3.5

### Refactor

- DRY getting project path
- Changed get paths to `PathBuf` usage

## [0.3.4] - 2023-03-07

### Documentation

- Added existing commands to README

### Features

- Implemented `delete` command, resolves #2

### Miscellaneous Tasks

- Update CHANGELOG.md for 0.3.4
- Bumped to 0.3.4

## [0.3.3] - 2023-03-01

### Miscellaneous Tasks

- Release rusmux version 0.3.3

## [0.2.0] - 2023-01-10

<!-- generated by git-cliff -->


completions/rusmux.bash
_rusmux() {
  local cur prev commands aliases projects

  COMPREPLY=()
  cur="${COMP_WORDS[COMP_CWORD]}"
  prev="${COMP_WORDS[COMP_CWORD - 1]}"

  commands="run stop debug edit delete new list copy doctor"
  aliases="start kill ls cp r s"

  if [[ "$prev" == "rusmux" ]]; then
    COMPREPLY=($(compgen -W "$commands $aliases" -- "$cur"))
    return 0
  fi

  projects=$(rusmux list 2>/dev/null)

  case "$prev" in
  run | start | r | s | stop | kill | debug | edit | delete)
    COMPREPLY=($(compgen -W "$projects" -- "$cur"))
    return 0
    ;;
  new)
    if [[ "$cur" == --* ]]; then
      COMPREPLY=($(compgen -W "--blank" -- "$cur"))
    else
      COMPREPLY=($(compgen -W "$projects" -- "$cur"))
    fi
    return 0
    ;;
  copy | cp)
    if [[ ${COMP_CWORD} -eq 2 ]]; then
      COMPREPLY=($(compgen -W "$projects" -- "$cur"))
    fi
    return 0
    ;;
  esac
}

complete -F _rusmux rusmux
